### ISSUE TITLE
Multicolored progress bar with transitional states not rendered correctly

### ISSUE THREAD

**Issue by sriyerg** [Rockhopper-Technologies/enlighten#41]
**Describe the bug**
I am attempting to implement a multicolored progress bar in a job scheduling type scenario where tasks get enqueued, after a while they start running and finally, the complete in either passed, failed or killed state. The enqueue is the main counter which is white, followed by sub-counters for running (yellow), passed (green) failed (red) and killed (magenta) states. The ideal outcome is at the end, the progress bar is devoid of white and yellow (all jobs terminate in passed failed or killed state). In the example I implemented, there is a single bar of white (queued) on the progress bar after it has finished, even though the queued and running counts are 0. 

**To Reproduce**
I modified the existing [examples/multicolored.py](https://github.com/Rockhopper-Technologies/enlighten/blob/2ebe9f4f89e03444b90a9524ee47b7876490218b/examples/multicolored.py#L116)  to make it look like this:

```python3

def load(manager, units=80):
    """
    Simulate some jobs.
    States are:
      queued: grey
      running: yellow
      passed: green
      failed: red
      killed: purple

      items from queued are moved over to running. a running test will either pass, fail or
      get killed.
    """

    terminal = manager.term
    bar_format = u'{desc}{desc_pad}{percentage:3.0f}%|{bar}| [' + \
                 terminal.white(u'Q:{count_0:{len_total}d}') + u' ' + \
                 terminal.yellow2(u'R:{count_1:{len_total}d}') + u' ' + \
                 terminal.green3(u'P:{count_4:{len_total}d}') + u' ' + \
                 terminal.red3(u'F:{count_3:{len_total}d}') + u' ' + \
                 terminal.magenta(u'K:{count_2:{len_total}d}') + u' ' + \
                 u'T:{total:{len_total}d}]'

    queued = manager.counter(total=units, desc='Building', unit='build',
                             color='white', bar_format=bar_format)
    running = queued.add_subcounter('yellow2')
    killed = queued.add_subcounter('magenta')
    failed = queued.add_subcounter('red3')
    passed = queued.add_subcounter('green3')

    queued_list = []
    running_list = []
    finished_list = []
    count = 0
  
    # Randomly pick pass (80%), failed (15%) or killed (5%) as the job status. 
    rand_status = ['P'] * 80 + ['K'] * 5 + ['F'] * 15

    while (passed.count + failed.count + killed.count) < units:
        time.sleep(random.uniform(0.05, 0.15))  # Random processing time

        for idx, node in enumerate(running_list):
            if node.loaded:
                running_list.pop(idx)
                finished_list.append(node)
                LOGGER.info('Job %d finished', node.iden)
                status = random.choice(rand_status)
                if status == 'P':
                    passed.update_from(running)
                if status == 'F':
                    failed.update_from(running)
                if status == 'K':
                    killed.update_from(running)

        for idx, node in enumerate(queued_list):
            if node.connected:
                queued_list.pop(idx)
                node.load()
                running_list.append(node)
                LOGGER.info('Job %d running', node.iden)
                running.update_from(queued)

        # Enqueue upto 5 jobs at a time.
        for _ in range(min(units - count, 5 - len(queued_list))):
            node = Node(count)
            node.connect()
            queued_list.append(node)
            LOGGER.info('Job %d enqueued', node.iden)
            queued.update()
            count += 1
```

**Environment (please complete the following information):**
 - Enlighten Version: 1.7.2
 - OS and version: Linux 
 - Console application: [e.g. xterm, cmd, VS Code Terminal] Gnome
 - Special Conditions: [e.g. Running under pyinstaller] None
 
![image](https://user-images.githubusercontent.com/46467186/107746702-1fa4ff80-6ccb-11eb-8a1b-70b26d222a08.png)

Please let me know if I did something obviously wrong here.

**Comment by avylove** [comment]
Thanks for reporting! In the future, please provide examples that can run without modification including all require imports.

This seems to be a rounding issue. Essentially, the code assumes any space left in the bar after calculating the subcounter bar lengths goes to the parent counter's bar. This rounding is done with int(), with the thought of avoiding overruns (0.5 and 0.5 would add to 2 with round()). I'll have to rework this logic a bit to account for 0 in the parent. I'll see if I can get that implemented.

Thanks again, this is a good find!

**Comment by sriyerg** [comment]
Will do, thanks! Really appreciate the quick response.

**Comment by avylove** [comment]
Fix is currently in master. Please test and let me know if it works for you.

**Comment by tom-brosch** [comment]
I had the same issue. Tested with master and it works for me now. Thanks for the fix. You're awesome.

**Comment by tom-brosch** [comment]
Some more feedback. The fix fixed the issue in most cases. However, there are very rare cases, where I still get the white bar, although the counter is zero. Here is an example

```python
import enlighten
import os

if __name__ == '__main__':

    print(f"Terminal size: {os.get_terminal_size()}")

    with enlighten.get_manager() as manager:
        terminal = manager.term
        bar_format = u'{desc}{desc_pad}{percentage:3.0f}%|{bar}| ' + \
                     u'W:{count_0:{len_total}d} ' + \
                     u'R:' + terminal.blue(u'{count_1:{len_total}d}') + ' ' + \
                     u'C:' + terminal.green(u'{count_2:{len_total}d}') + ' ' + \
                     u'F:' + terminal.red(u'{count_3:{len_total}d}') + ' ' + \
                     u'A:' + terminal.yellow(u'{count_4:{len_total}d}') + ' |'

        description = "Some description                                       "
        waiting = manager.counter(total=51, desc=description, unit='jobs', color='gray', bar_format=bar_format)
        running = waiting.add_subcounter('blue')
        completed = waiting.add_subcounter('green')
        failed = waiting.add_subcounter('red')
        canceled = waiting.add_subcounter('yellow')

        waiting.update(37)
        running.update_from(waiting, 1)
        completed.update_from(waiting, 36, force=True)
```

Which produces the following output:

![image](https://user-images.githubusercontent.com/71647919/108323572-21892a00-71c7-11eb-8427-819d084ab887.png)

Changing the width of the terminal or changing the length of the descriptions makes the issue disappear. So far, I was only able to reproduce the issue with these exact value for therminal width, counters and description length. Tested with enlighten from master on OpenSUSE with Python 3.6.

**Comment by avylove** [comment]
I was able to reproduce. Let me see if i can figure out what's going on. Thanks!

**Comment by avylove** [comment]
Fix is in master. Please let me know if you see any other issues.

**Comment by tom-brosch** [comment]
Can confirm fix. Thanks a lot.

**Comment by avylove** [comment]
Fix in 1.8.0.

@sriyerg Thanks for reporting!
@tom-brosch Thanks for testing!
